spec/spec_tests/uri_options_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'lite_spec_helper'
+
+describe 'Uri Options' do
+  include Mongo::ConnectionString
+
+  URI_OPTIONS_TESTS.each do |file|
+
+    spec = Mongo::ConnectionString::Spec.new(file)
+
+    context(spec.description) do
+
+      before(:all) do
+
+        module Mongo
+          class Address
+
+            private
+
+            alias :original_create_resolver :create_resolver
+            def create_resolver(timeout, ssl_options)
+              family = (host == 'localhost') ? ::Socket::AF_INET : ::Socket::AF_UNSPEC
+              info = ::Socket.getaddrinfo(host, nil, family, ::Socket::SOCK_STREAM)
+              FAMILY_MAP[info.first[4]].new(info[3], port, host)
+            end
+          end
+        end
+      end
+
+      after(:all) do
+
+        module Mongo
+          # Return the implementations to their originals for the other
+          # tests in the suite.
+          class Address
+            alias :create_resolver :original_create_resolver
+            remove_method(:original_create_resolver)
+          end
+        end
+      end
+
+      spec.tests.each do |test|
+
+        # Skip optional tests for options the driver doesn't support.
+        next if test.description.start_with?('tlsAllowInvalidHostnames', 'tlsInsecure')
```

Comment:
If tlsInsecure is now supported by the driver, why is its test skipped?

Reply:
Fixed